UnitTests/Schemas/IFC4X1/extruded-solid/src/extruded-solid.cpp

```diff
@@ -96,65 +83,202 @@ TEST_F(ExtrudedSolidTest, ImageIsSaved)
 {
-	renderer->setViewDirection(buw::eViewDirection::Front);
-	compareImageWithView("extruded-solid_front.png");
-}
+	// Arrange
+	const auto expected_front_left_bottom = buw::loadImage4b(dataPath("extruded-solid_front_left_bottom.png").string());
+	const auto expected_front_right_bottom = buw::loadImage4b(dataPath("extruded-solid_front_right_bottom.png").string());
+	const auto expected_top_left_front = buw::loadImage4b(dataPath("extruded-solid_top_left_front.png").string());
+	const auto expected_top_front_right = buw::loadImage4b(dataPath("extruded-solid_top_front_right.png").string());
+	const auto expected_top_left_back = buw::loadImage4b(dataPath("extruded-solid_top_left_back.png").string());
+	const auto expected_top_right_back = buw::loadImage4b(dataPath("extruded-solid_top_right_back.png").string());
+	const auto expected_back_left_bottom = buw::loadImage4b(dataPath("extruded-solid_back_left_bottom.png").string());
+	const auto expected_right_bottom_back = buw::loadImage4b(dataPath("extruded-solid_right_bottom_back.png").string());
+
+	// Act (FrontLeftBottom)
+	renderer->setViewDirection(buw::eViewDirection::FrontLeftBottom);
+	buw::Image4b image_front_left_bottom = CaptureImage();
+	// Act (FrontRightBottom)
+	renderer->setViewDirection(buw::eViewDirection::FrontRightBottom);
+	buw::Image4b image_front_right_bottom = CaptureImage();
+	// Act (TopLeftFront)
+	renderer->setViewDirection(buw::eViewDirection::TopLeftFront);
+	buw::Image4b image_top_left_front = CaptureImage();
+	// Act (TopFrontRight)
+	renderer->setViewDirection(buw::eViewDirection::TopFrontRight);
+	buw::Image4b image_top_front_right = CaptureImage();
+	// Act (TopLeftBack)
+	renderer->setViewDirection(buw::eViewDirection::TopLeftBack);
+	buw::Image4b image_top_left_back = CaptureImage();
+	// Act (TopRightBack)
+	renderer->setViewDirection(buw::eViewDirection::TopRightBack);
+	buw::Image4b image_top_right_back = CaptureImage();
+	// Act (BackLeftBottom)
+	renderer->setViewDirection(buw::eViewDirection::BackLeftBottom);
+	buw::Image4b image_back_left_bottom = CaptureImage();
+	// Act (RightBottomBack)
+	renderer->setViewDirection(buw::eViewDirection::RightBottomBack);
+	buw::Image4b image_right_bottom_back = CaptureImage();
 
-TEST_F(ExtrudedSolidTest, GivenNewImage_AfterHome_AreEqual_Bottom)
-{
-	renderer->setViewDirection(buw::eViewDirection::Bottom);
-	compareImageWithView("extruded-solid_bottom.png");
-}
+	// Assert
+	EXPECT_EQ(image_front_left_bottom, expected_front_left_bottom);
+	EXPECT_EQ(image_front_right_bottom, expected_front_right_bottom);
+	EXPECT_EQ(image_top_left_front, expected_top_left_front);
+	EXPECT_EQ(image_top_front_right, expected_top_front_right);
+	EXPECT_EQ(image_top_left_back, expected_top_left_back);
+	EXPECT_EQ(image_top_right_back, expected_top_right_back);
+	EXPECT_EQ(image_back_left_bottom, expected_back_left_bottom);
+	EXPECT_EQ(image_right_bottom_back, expected_right_bottom_back);
+
+	// Annihilate
 
+}
```

Comment:
Is this a mere duplication of the other functions? If yes, you can delete these, since you are checking the same thing twice.

Reply:
Resolved: 16c401a